### DISCUSSION TITLE
ADMaterial Problems

### DISCUSSION THREAD

**Discussion by YuhangNIU** [idaholab/moose#17822] [Q&A General] [answered]
Hi all，

I know the old and older material property values can be obtained through "getMaterialPropertyOld and getMaterialPropertyOlder" in non-AD Material. However, could anyone kindly tell me how to get the AD material property value in ADMaterial at the previous time step?

Thanks,
Yuhang

**Comment by hugary1995** [comment, marked as answer]
`getMaterialPropertyOld` and `getMaterialPropertyOlder` work for AD material properties as well.

There is no point in getting old and older material properties with their dual numbers, because all dual numbers will be zero.
